src/NuGet.Services.AzureSearch/SearchService/AzureSearchQuery.cs

```diff
@@ -0,0 +1,161 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace NuGet.Services.AzureSearch.SearchService
+{
+    /// <summary>
+    /// The type used to generate Azure Search Service queries. Used by <see cref="SearchTextBuilder"/>.
+    /// Given the query "fieldA:value1 value2":
+    /// 
+    ///   * "value1" is a field-scoped value 
+    ///   * "value2" is a non-field scoped value
+    /// </summary>
+    internal class AzureSearchQuery
```

Comment:
Maybe `AzureSearchQueryBuilder` to emphasize the statefulness.

Reply:
Renamed